gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java

```diff
@@ -280,8 +291,15 @@ private boolean ensureHiveTableExistenceBeforeAlternation(String tableName, Stri
         try (Timer.Context context = this.metricContext.timer(GET_HIVE_TABLE).time()) {
           existingTable = HiveMetaStoreUtils.getHiveTable(client.getTable(dbName, tableName));
         }
+        HiveTable schemaSourceTable = existingTable;
+        if (state.contains(SCHEMA_SOURCE_DB) || state.contains(SCHEMA_SOURCE_TABLE)) {
```

Comment:
Should the information of source DB coming from HiveSpec instead of `state`?  IIRC `state` will be applied for all paths' registration here, it is more like a global state but not for a specific table/db

Reply:
In both compaction pipeline and ingestion pipeline, this state is topic specific state. And for this case that db is registered as an additional db, the db name is specified by a job level config. So I think for all the tables in this db, it should has the same source db. That's why I do want to make this config a job level config also.